drivers/adc/adc_mcp320x.c

```diff
@@ -280,25 +275,19 @@ static int mcp320x_init(const struct device *dev)
 	data->dev = dev;
 
 	k_sem_init(&data->sem, 0, 1);
-	data->spi_dev = device_get_binding(config->spi_dev_name);
 
-	if (!data->spi_dev) {
-		LOG_ERR("SPI master device '%s' not found",
-			config->spi_dev_name);
+	if (!device_is_ready(config->spi_dev)) {
+		LOG_ERR("SPI master device '%s' not ready",
+			log_strdup(config->spi_dev->name));
```

Comment:
As in https://github.com/zephyrproject-rtos/zephyr/pull/35741#pullrequestreview-670981411, why duplicate this name?

Reply:
Thanks, force of habit when seeing a `%s` in a `LOG_*()` macro. Fixed.